### PR TITLE
feat(trace): Default to all projects for traces

### PR DESCRIPTION
**Pull request by wmak**
- This adds an include_all_projects to get_snuba_params which works the same way as if the user had passed `project=-1`
- This is cause we always want to load all the projects for a trace since a trace can span any number of projects in a given org

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events_trace.py [hunk 1]
```diff
@@ -817,7 +817,9 @@ def get(self, request: HttpRequest, organization: Organization, trace_id: str) -
 
         try:
             # The trace view isn't useful without global views, so skipping the check here
-            params = self.get_snuba_params(request, organization, check_global_views=False)
+            params = self.get_snuba_params(
+                request, organization, check_global_views=False, include_all_projects=True
+            )
```
[the remark this answers is not in the document]
Comment (wmak) [reply src/sentry/api/endpoints/organization_events_trace.py:822]: 🤔 Hmm I wanted to keep that behaviour intentionally, this basically changes the default behaviour from doing `my projects` to doing `all projects` but hopefully maybe eventually 🙏 one day we can narrow projects down somehow? 🙃

I'll add a test though :+1:

**Review by narsaynorath — APPROVED** [review]
lgtm

File: src/sentry/api/endpoints/organization_events_trace.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Zylphrex) [reply src/sentry/api/endpoints/organization_events_trace.py:822]: What's the reason to keep this behaviour? This opens the endpoint to the opportunity for bugs if the frontend specifies too few projects right?

File: src/sentry/api/endpoints/organization_events_trace.py [hunk 2]
```diff
@@ -743,6 +744,15 @@ class OrganizationEventsTraceEndpointBase(OrganizationEventsV2EndpointBase):
...
     }
 
+    def get_projects(self, request: Request, organization, project_ids=None, project_slugs=None):
+        return super().get_projects(
+            request,
+            organization,
+            project_ids={-1},
+            project_slugs=None,
+            include_all_accessible=True,
+        )
```
Comment (Zylphrex) [inline src/sentry/api/endpoints/organization_events_trace.py:754]: A comment explaining why this override is necessary will be helpful here as it's not immediately obvious why this is needed.

**Review by Zylphrex — APPROVED** [review]

**Pull request merged**